backend/person-service/src/main/kotlin/com/nosql/personservice/component/impl/DefaultPersonComponent.kt

```diff
@@ -24,27 +29,133 @@ class DefaultPersonComponent(
-        val operationDetails = "Save 'person' record with id = ${person.id}"
+        val operationDetails = "Save 'person' record with id = ${person.id.toHexString()}"
 
         log.logBefore(operationDetails)
 
+        assertUserNotExists(person)
+
         return personRepository.save(person)
             .onErrorMap { handleError(it, operationDetails) }
             .doOnSuccess { log.logSuccess(operationDetails) }
             .awaitSingle()
     }
 
-    override suspend fun delete(userObjectId: ObjectId) {
-        val operationDetails = "Delete 'person' record with id = ${userObjectId.toHexString()}"
+    override suspend fun get(personId: ObjectId): PersonEntity {
+        val operationDetails = "Get 'person' record with id = ${personId.toHexString()}"
 
         log.logBefore(operationDetails)
 
-        personRepository.deleteById(userObjectId)
+        return personRepository.findById(personId)
+            .switchIfEmpty(NotFoundException(description = "Not found 'person' record by id = '$personId'").toMono())
+            .onErrorMap { handleError(it, operationDetails) }
+            .doOnSuccess { log.logSuccess(operationDetails) }
+            .awaitSingle()
+    }
+
+    override suspend fun getAll(pageable: Pageable): List<PersonEntity> {
+        val operationDetails = "Get 'person' records"
+
+        log.logBefore(operationDetails)
+
+        return personRepository.findByIdNotNull(pageable)
+            .onErrorMap { handleError(it, operationDetails) }
+            .collectList()
+            .doOnSuccess { log.logSuccess(operationDetails) }
+            .awaitSingle()
+    }
+
+    override suspend fun getByName(name: String, pageable: Pageable): List<PersonEntity> {
+        val operationDetails = "Get 'person' records with name like '$name'"
+
+        log.logBefore(operationDetails)
+
+        return personRepository.findAllByNameIsLike(name, pageable)
+            .onErrorMap { handleError(it, operationDetails) }
+            .collectList()
+            .doOnSuccess { log.logSuccess(operationDetails) }
+            .awaitSingle()
+    }
+
+    override suspend fun extendedGet(
+        name: String,
+        surname: String,
+        patronymic: String,
+        sex: String,
+        position: String,
+        status: String,
+        startAge: Date,
+        endAge: Date,
+        pageable: Pageable,
+    ): List<PersonEntity> {
+        val operationDetails = "Get 'person' records with name like '$name', surname like '$surname', patronymic " +
+                "like '$patronymic', sex like '$sex', position like '$position', status like '$status', startAge " +
+                "like '$startAge', endAge like '$endAge'"
+
+        log.logBefore(operationDetails)
+
+        return personRepository.findExtended(
+            name,
+            surname,
+            patronymic,
+            sex,
+            position,
+            status,
+            startAge,
+            endAge,
+            pageable
```

Comment:
use kotlin named argument, because we may accidentally swap the parameters
```suggestion
            name = name,
            surname = surname,
            ...
```

Reply:
fixed